File: 프로그래머스/lv2/42578. 위장/위장.py
def solution(clothes):
    # format clothes into wardrobe : key is 2nd string(type)
    hash_map = {}
    for name, type in clothes:
        # initialize hashmap with 1 for case where spy chooses only one item
        hash_map[type] = hash_map.get(type, 1) + 1
    # multiply each occurences of types to get all combinations
    answer = 1
    for type in hash_map:
        answer *= hash_map[type]
    # return case where spy isn't wearing anything
    return answer - 1

File: jeonpilseong/프로그래머스/lv2/42578. 위장/위장.js
function solution(clothes) {
    let dict = {};
    for(let i of clothes){
        !(i[1] in dict) ? dict[i[1]] = 2 : dict[i[1]] += 1
    }
    
    let cnt = 1;
    for(let i in dict){
        cnt *= dict[i];
    }
    
    return cnt - 1;
}
